capella2polarion/converters/converter_config.py

```diff
@@ -39,20 +43,98 @@ class LinkConfig:
+                    polarion_role=add_prefix(
+                        (pid := link.get("polarion_role", lid)),
+                        role_prefix,
+                    ),
+                    include=link.get("include", {}),
+                    link_field=(lf := link.get("link_field", pid)),
+                    reverse_field=link.get("reverse_field", f"{lf}_reverse"),
+                )
+            else:
+                logger.error(  # type: ignore[unreachable]
+                    "Link not configured correctly: %r",
+                    link,
+                )
+                continue
+            result.append(config)
+        return result
+
 
 @dataclasses.dataclass
 class CapellaTypeConfig:
     """A single Capella Type configuration."""
 
     p_type: str | None = None
-    converters: str | list[str] | dict[str, dict[str, t.Any]] | None = None
+    converters: dict[str, dict[str, t.Any]] | None = None
     links: list[LinkConfig] = dataclasses.field(default_factory=list)
     is_actor: bool | None = None
     nature: str | None = None
 
-    def __post_init__(self):
-        """Post processing for the initialization."""
-        self.converters = _force_dict(self.converters)
+    @staticmethod
+    def _force_dict(
+        converters: str | list[str] | ConvertersType | None,
+    ) -> dict[str, dict[str, t.Any]]:
+        match converters:
+            case None:
+                return {}
+            case str():
+                return {converters: {}}
+            case list():
+                return {c: {} for c in converters}
+            case dict():
+                return CapellaTypeConfig._filter_config(converters)
+            case _:
+                raise TypeError("Unsupported Type")
+
+    @staticmethod
+    def _filter_config(
+        converters: dict[str, t.Any],
+    ) -> dict[str, dict[str, t.Any]]:
+        custom_converters = (
+            "include_pre_and_post_condition",
+            "linked_text_as_description",
+            "add_attributes",
+            "add_context_diagram",
+            "add_tree_diagram",
+            "add_jinja_fields",
+            "jinja_as_description",
+        )
+        filtered_config = {}
+        assert isinstance(converters, dict)
+        for name, params in converters.items():
+            params = params or {}
+            if name not in custom_converters:
+                logger.error("Unknown converter in config %r", name)
+                continue
+
+            if name in ("add_context_diagram", "add_tree_diagram"):
+                assert isinstance(params, dict)
+                params = _filter_context_diagram_config(params)
+
+            if name in ("add_attributes",):
+                assert isinstance(params, list)  # type: ignore[unreachable]
+                params = {"attributes": params}  # type: ignore[unreachable]
+
+            filtered_config[name] = params
+
+        return filtered_config
```

Comment:
Removed the post-init processing which makes the type annotation of `converters` unsolveable (therefore useless for mypy). I moved the `_force_config` functions as staticmethods to the respective class. They need to be used on their parameter before instantiating the object. Less automation but clear type annotation.